forms/OpenIDPluginSettingsForm.php

```diff
@@ -0,0 +1,237 @@
+
+    /**
+     * @copydoc Form::initData()
+     */
+    function initData()
+    {
+        $request = Application::get()->getRequest();
+        $contextId = ($request->getContext() == null) ? 0 : $request->getContext()->getId();
+        $settingsJson = $this->plugin->getSetting($contextId, 'openIDSettings');
+        $settings = json_decode($settingsJson, true);
+        $provider = $settings['provider'];
+
+        if ($provider && is_array($provider)) {
+            foreach ($provider as &$prov) {
+                if (!empty($prov['clientId'])) {
+                    $prov['clientId'] = self::HIDDEN_CHARS;
+                }
+                if (!empty($prov['clientSecret'])) {
+                    $prov['clientSecret'] = self::HIDDEN_CHARS;
+                }
+            }
+        }
+        if (isset($settings)) {
+            $this->_data = [
+                'initProvider' => OpenIDPlugin::PUBLIC_OPENID_PROVIDERS,
+                'provider' => $provider,
+                'legacyLogin' => $settings['legacyLogin'] ?? true,
+                'legacyRegister' => $settings['legacyRegister'] ?? true,
+                'disableConnect' => $settings['disableConnect'] ?? false,
+                'hashSecret' => $settings['hashSecret'],
+                'generateAPIKey' => $settings['generateAPIKey'] ?? 0,
+                'providerSync' => $settings['providerSync'] ?? false,
+                'disableFields' => $settings['disableFields'],
+            ];
+        } else {
+            $this->_data = [
+                'initProvider' => OpenIDPlugin::PUBLIC_OPENID_PROVIDERS,
+                'legacyLogin' => true,
+                'legacyRegister' => false,
+                'generateAPIKey' => false,
+            ];
+        }
+        parent::initData();
+    }
+
+    /**
+     * @copydoc Form::readInputData()
+     */
+    function readInputData()
+    {
+        $this->readUserVars(
+            [
+                'provider',
+                'legacyLogin',
+                'legacyRegister',
+                'disableConnect',
+                'hashSecret',
+                'generateAPIKey',
+                'providerSync',
+                'disableFields'
+            ]
+        );
+        parent::readInputData();
+    }
+
+    /**
+     * @copydoc Form::fetch()
+     */
+    public function fetch($request, $template = null, $display = false)
+    {
+        $context = $request->getContext();
+        $redirectURL = $context == null ? $request->getIndexUrl().'/openid/doAuthentication' : $request->getIndexUrl().'/'.$context->getPath(
+            ).'/openid/doAuthentication';
+        $redirectMSURL = $context == null ? $request->getIndexUrl().'/openid/doMicrosoftAuthentication' : $request->getIndexUrl().'/'.$context->getPath(
+            ).'/openid/doMicrosoftAuthentication';
+        $templateMgr = TemplateManager::getManager($request);
+        $request->getBasePath();
+        $templateMgr->assign('pluginName', $this->plugin->getName());
```

Comment:
Use `$templateMgr->assign(['a' => $a, ... ])` syntax for batch assignments.

Reply:
Changed to 

```
$templateMgr->assign([
    'pluginName' => $this->plugin->getName(),
    'redirectUrl' => $redirectURL,
    'redirectMSUrl' => $redirectMSURL
]);
```